addons/account_edi/models/account_edi_format.py

```diff
@@ -455,3 +455,38 @@ def _retrieve_currency(self, code):
         :returns:    A currency or an empty recordset if not found.
         '''
         return self.env['res.currency'].search([('name', '=', code.upper())], limit=1)
+
+    def _retrieve_bank_account(self, acc_number, partner):
+        '''Search all bank accounts, including the inactive ones. If create is True and
+        the bank account is not found, one is created and set to inactive.
+        It should not be set to active until confirmation from the user that it is valid.
+
+        :param acc_number: The account number to find.
+        :param partner:    The partner to wich this account belongs.
+        :returns:          The bank account or an empty recordset if not found.
+        '''
+        bank_account = self.env['res.partner.bank'].search([
```

Comment:
Yeah ok. I'm not so sure about the best way to implement it. If I merge them, we would have the same discussion that we previously had with the "create" parameter (that I found useful because, as this is a toolbox to be used by child edi formats, I think it's good to give the possibility to search without creating).

But we would have an additional problem : we would need to know whether the account was created or not, because if it wasn't we assign it to `move.partner_bank_id`, but if it was, we can't do that until the user has explicitly validated it.

On the other hand, I'm not too keen about the assigning part to `imported_bank_id` on this method because if the account exists, the assignment to `partner_bank_id` is done by the edis which creates unbalance. I realized that when trying to write a nice docstring.

So to have a solution that is simple and keeps the flow balanced (easier to understand) here's what I did :
- Merge the methods, and create an inactive bank account if it's not found.
- Keep the `create` parameter for flexibility in future implementation of new edis.
- The assignment to `imported_bank_id` (and `partner_bank_id`) is the responsability of the EDI and is based on whether the returned bank account is active or not. That means that if an EDI calls the method `_retrieve_bank_account` with `create=True` and do nothing with the result, an inactive bank_account is created but it will not show in the account_move (it will be useless).
- To avoid this, the default value of `create` is False. Only a developer that has read the docstring will use this parameter. 

Let me know what you think. I'm open to any suggestions to improve this flow.

Reply:
Also, we could the assignment part of both `imported_bank_id ` and `partner_bank_id` in the helper since this code is duplicated in all the EDIs and will most likely be used again, but I'm not so sure about that either since it's not how all the `retreive_something` helpers work, creating new unbalances and subtilties to understand. Unless we make it **very** explicit (in the method name and in the docstring).

Reply:
I'm okay with the changes to _retrieve_bank_account. 

Indeed, it's a pity to have this code duplicated in all EDIs. I think we could create a new helper for that, called something like _retrieve_and_assign_bank_account, and calling _retrieve_bank_account.